controllers/machinehealthcheck_targets.go

```diff
@@ -0,0 +1,256 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	machinePhaseFailed = "Failed"
+
+	// Timeout waiting for the provider to bring up a node
+	timeoutForMachineToHaveNode = 10 * time.Minute
```

Comment:
We might want to make this part of the API/annotation since this timeframe might be different for different types of instances and different providers

Reply:
+1 to making this configurable. Prefer to bias towards adding to the api type rather than annotation, though.

Reply:
I've added this as an API field, though I'm not sure about the name I've used for it, `NodeStartupTimeout` 🤔
I added a default value for the field in the webhook and in the controller in case for some reason the field is nil in the controller too, does that make sense to do?

Reply:
`NodeStartupTimeout` sounds reasonable to me, not sure if others have other suggestions cc @bison.

@JoelSpeed Let's make sure we put a PR to update the proposal appropriately after this PR merges.

Reply:
I don't have a strong opinion -- `NodeStartupTimeout` seems fine, but don't the terms Provisioning and Provisioned denote this transition elsewhere in the API? Maybe `ProvisioningTimeout` or something similar is clearer?

Reply:
I think I'd lean towards keeping `NodeStartupTimeout` in that it is explicit that it is waiting for the Node, vs Provisioning, which is more generally tied to the concept of a Machine.